elk/extraction/prompt_dataset.py

```diff
@@ -222,12 +224,12 @@ def __init__(self, cfg: PromptConfig):
                     "few-shot prompts"
                 )
 
-            self.fewshot_strata = [
+            self.fewshot_strata: list[Dataset] = [
                 ds_dict["train"].filter(lambda ex: ex[label_column] == i)
                 for i in range(self.num_classes)
             ]
         else:
-            self.fewshot_strata: list[Dataset] = []
+            self.fewshot_strata = []
```

Comment:
otherwise mypy will complain that the type already got declared. so the first branch must be the first the declare it.

Reply:
> @thejaminator Maybe we should just introduce a wrapper around `typing.cast` that also runtime-asserts the right type? That would actually be cleaner than what we're doing right now in a lot of places (sticking an assert statement right after a declaration). If it's supported by both Mypy and Pyright that'd be great.

ah yeah, basically the assert_is_instance i implement here https://github.com/EleutherAI/elk/pull/93/files/245859cb3d5a74e569e0800c191ee7ef4a02feb7..51db6995fe583d29ae5e3a863fda75724af6ce76#r1113128336